src/commands/unset.ts

```diff
@@ -0,0 +1,27 @@
+import { ExecuteResult } from "../../mod.ts";
+import { CommandContext } from "../command_handler.ts";
+import { resultFromCode } from "../result.ts";
+
+export function unsetCommand(context: CommandContext): ExecuteResult {
+  try {
+    return {
+      kind: "continue",
+      code: 0,
+      changes: parseNames(context.args).map((name) => ({ kind: "unsetvar", name })),
+    };
+  } catch (err) {
+    context.stderr.writeLine(`unset: ${err?.message ?? err}`);
+    return resultFromCode(1);
+  }
+}
+
+function parseNames(args: string[]) {
+  if (args[0] === "-f") {
+    // we don't support the -f (function) flag
+    throw Error(`unsupported flag: -f`);
+  } else if (args[0] === "-v") {
+    return args.slice(1);
+  } else {
+    return args;
+  }
+}
```

Comment:
Maybe we need to check if `args` is not empty before accessing `[0]`?

Reply:
It would be better to be more explicit. The Rust implementation definitely should. I’ll add a test tomorrow for no args.